project/Build.scala

```diff
@@ -41,7 +41,8 @@ object SummingbirdBuild extends Build {
   val novocodeJunitVersion = "0.10"
   val specs2Version = "1.13"
 
-  val sparkCoreVersion ="0.9.0-incubating"
+  val sparkCoreVersion ="1.2.0"
+  val scalazVersion = "7.1.1"
```

Comment:
please use the Algebird implementation of this since we already depend on Algebird.

com.twitter.algebird.monad.Reader and the code may just compile.